tests/csl-json/test_csl_pruning.py

```diff
@@ -0,0 +1,20 @@
+import json
+import pathlib
+
+import pytest
+
+from manubot.cite.citeproc import remove_jsonschema_errors
+
+directory = pathlib.Path(__file__).parent
+csl_instances = [
+    x.name for x in directory.glob('*-csl')
+]
+
+
+@pytest.mark.parametrize('name', csl_instances)
+def test_remove_jsonschema_errors(name):
```

Comment:
These tests cover most of the types of CSL JSON files Manubot would receive from its typical sources.  They don't tell us how pruning will behave one other sources.  That should be okay, but I'm thinking about whether we want to add a comment for maintenance.  If Manubot adds new metadata sources, those should probably have their pruning tested here as well.

Do we need to test pruning a URL reference?

Reply:
> Do we need to test pruning a URL reference?

Yes.

>  If Manubot adds new metadata sources, those should probably have their pruning tested here as well.

Definitely! I can see where the best place to document this is.

Reply:
Note it's not as simple as one test per metadata source. For example, DOI content negotiation returns different raw CSL based on the registrar. Also we're taking the output at a single point in time. These metadata endpoints change their responses over time. For example, previously Crossref didn't list all the references. Now it does.

Reply:
The new test and docstring are very helpful.